### PR TITLE
Display a security badge with the installation status

### PR DESCRIPTION
**Pull request by crohr**
This uses the Release API to determine the status of the current OpenProject installation, and generate an SVG badge that is displayed in the admin block on the welcome page.

Available status:
* vulnerable
* unknown
* up to date
* update available

TODO:
- [x] update packager postinstall to set `OPENPROJECT_INSTALLATION__TYPE` to `packager`
- [x] send whether valid EE token in use

### CODE REVIEW

**Comment by oliverguenther** [comment]
LGTM! We will still need a way to set a setting (installation type) from command line since we can only set configuration values for now.

**Comment by crohr** [comment]
@oliverguenther thanks for the review, and yes this PR still needs:

* a way to update the security_badge_displayed boolean from the admin settings
* a change to our docker image and packages to set the `OPENPROJECT_INSTALLATION__TYPE` config value
* a few tests if possible

**Comment by ulferts** [comment]
As the badge is currently displayed only on the welcome page it might get overlooked quite often as the page is not visited that much. Speaking only for myself, I never visit the page anymore. I would therefor never be informed of a pending update.

An alternative would be to have the badge loaded on every page (similar to what we do for the unsupported browser error message) and, in order to not be too nagging, to offer the ability to hide the message for that particular update version (also similar to how we handle the unsupported browser error message).

(The change in the error message is handled in #7029)

**Comment by oliverguenther** [comment]
I like the idea with the browser error message :+1:

**Comment by crohr** [comment]
I like the idea too, but that's potentially two notification headers that get displayed, and then if snoozed or permanently hidden for that release, that installation goes missing in stats, and the user probably forgets to check the install status by other means.

I would be more inclined to display the badge on more pages, but only for admins.

**Comment by crohr** [comment]
After some discussion, for this first pass we'll display the badge on the admin block in the welcome page, and close to the version number in the admin section.

I'll also alter the badge URL so that it sends the language currently used by the user, in case we want to generate language-aware badges later.

**Comment by crohr** [comment]
This is what it looks like when put in the information admin panel:

image

**Comment by crohr** [comment]
Question: what do we want to do on the SaaS platform? Doesn't make sense to display the badge right? Can we force a setting based on an env variable?

**Comment by oliverguenther** [comment]
> Can we force a setting based on an ENV variable?

Not that I know of, but we can seed the setting in the multitenancy plugin

**Comment by crohr** [comment]
@oliverguenther right, makes sense thanks

**Comment by crohr** [comment]
Question: Do I need to update the country-specific locales? How does it work?

File: spec/helpers/security_badge_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+require 'spec_helper'
+
+describe SecurityBadgeHelper, type: :helper do
+  describe '#security_badge_url' do
+    before do
+      Setting.installation_uuid = "abcd1234"
```
Comment (oliverguenther) [inline spec/helpers/security_badge_helper_spec.rb]: prefer: 

```
describe '#security_badge_url', with_settings: { installation_uuid: 'abcd1234' } do
```

Comment (crohr) [reply spec/helpers/security_badge_helper_spec.rb]: Thanks!

**Review by oliverguenther — APPROVED** [review]
LGTM and works fine. Feel free to merge before or after adding the EE check

**Pull request merged**